### DISCUSSION TITLE
Provide better documentation

### DISCUSSION THREAD

**Discussion by uralets** [dosemu2/dosemu2#1604] [General]
Specifically (but not limited to) on the topics such as:
General operation
Initialization
ALSA MIDI configuration
Maybe remove the old dosemu1 docs too - it's confusing.

Almost no docs seem to get shipped with the Ubuntu PPA packages. The README.html seems to be carried over from dosemu1.

I've decided to give dosemu2 a try playing Terra Nova (https://github.com/dosemu2/dosemu2/issues/1599 and https://github.com/dosemu2/dosemu2/issues/1600) and found myself confronted with a couple of issues and started digging for documentation and sadly found out that almost no docs are shipped with the pre-built packages neither anything was found online (in the vein of https://docs.scummvm.org/en/latest/)

For example I've tried launching DOSEMU2 without FDPP (btw how could it be disabled while it's installed? Would be nice to document this too ) bc the game mentioned above is behaving badly - and since I'm new to the fact that DOSEMU2 now has something like FDPP I've decided to experiment and use a FreeDOS installation (when I realized that FDPP sort of "replaces" it) but there's no "quick start" or anything like that and DOSEMU2 simply doesn't work out of the box without FDPP. BTW the PPA package doesn't depend on it - thus a user upon installing the package is left with a dysfunctional piece of software and virtually no clear instructions on what to do next.
DOSEMU2 is an emulator OR a virtual machine - that much is clear. And like other VMs we could try and properly install a FreeDOS inside it - from the official install media.. but should we really go through these chores for some DOS experience? You would agree it's a bit involved compared to DOSBOX.

**I do realize now having poked around that FDPP seems "the official" way to go** and I'm ofc OK with that, but you would agree the whole thing needs some high-level overview/documentation.

The Python scripts that mention FreeDOS in their names ARE getting installed but again no description how why and what for you would use them. They do seem to work (see below tho) - like download some FreeDOS-related files into .cache/ and the other one copies them into .dosemu/drive_c but still it's a mystery what exactly is going on.
Also I think I've found a bug in one of them dosemu2/install-freedos#12

As for ALSA MIDI - I have a soundcard with a real hardware MIDI (Audigy ZS 2 or something) and would like to use it for MIDI output. I do so successfully with other programs like DOSBOX and SCUMMVM (or aplaymidi for that matter).
Again, it's not clear what do you use - alsa_midi or alsa_virmidi and what's the difference between the two. Both are mentioned in the supplied dosemu.conf. But that's it. Specifically I need to point DOSEMU2 to the correct MIDI port seen in the output of aplaymidi -l. How do I do it? 
When using eg aplaymidi to play a midi file I can see

client 128: 'aplaymidi' [type=user,pid=11037]
    0 'aplaymidi       '
	Connecting To: 17:0

in the output of aconnect -l

OTOH while DOSEMU2 is working it shows
client 128: 'Client-128' [type=user,pid=11055]
    0 'Virtual RawMIDI '

Setting $_midi_driver = "alsa_midi" set in ~/.dosemurc doesn't seem to change the above output..
Grepping the sources for anything ALSA MIDI related here on github hasn't yielded anything useful... It did though made me stumble on various DOSEMU1 config bits and pieces - no idea if those are even used now?

DOSBOX-staging plays MIDI music in Terra Nova fine when the latter is configured to use General Midi (midi_num 12 in TN.CFG)
And if I clear the soundfont bank with asfxload -i the music stops playing - as it should. While in DOSEMU2 I don't hear it at all.

It seems a lot of effort is going into this project but the somewhat haphazard documentation IMHO hampers it.

I sure would like to play some DOS games with significantly less strain on the CPU - something that DOSBOX(-staging) authors has never seen as a priority (or even a possibility) AFAIK.

**Comment by stsp** [comment]
> Maybe remove the old dosemu1 docs too - it's confusing.

Yes, that should be done.

> The Python scripts that mention FreeDOS in their names ARE getting installed but again no description how why and what for 

But there is:
https://github.com/dosemu2/dosemu2/wiki/installing-various-DOSes
Our wiki is still very poorly structured and
has a notably bad articles too, but here it is.

> It seems a lot of effort is going into this project but the somewhat haphazard documentation IMHO hampers it.

People who seek the documentation, are
just looking for problems. dosemu2 should
work out of the box. Which is why the documentation
is not within, but in a wiki. The user should start
w/o any docs and just _try_ things out. If he doesn't
try and instead is looking for a documentation, he
already lost the battle. :) But as soon as he tried
and made sure the basic things work, then he may
start wandering around wiki for some advanced topics.

> I sure would like to play some DOS games with significantly less strain on the CPU

Not sure if you can have a good power saving
with Terra Nova game on dosemu2 any soon,
sorry about that. We can try to make it run faster
(though I don't see sluggishness that you mention)
but power saving is not a priority right now.

> DOSBOX-staging plays MIDI music in Terra Nova fine when the latter is configured to use General Midi

Same with dosemu2 - without any configuration
or documentation. It just works, I have checked.

**Comment by stsp** [comment]
I mean, just how much documentation do
you read before starting firefox? I guess none,
it just works. Only when it doesn't, you go
to google. But in most cases you end up in
KB or bugzilla, not in a documentation, anyway.

**Comment by stsp** [comment]
> For example I've tried launching DOSEMU2 without FDPP (btw how could it be disabled while it's installed?

If there is some other DOS on drive
C, then it boots. If there is some other
command.com on drive C, then its used.
fdpp or comcom32 never come ahead
of anything you copied to drive C.
The aforementioned wiki page mentions
that fact.
Any documentation that you see missing,
can be written by yourself on our wiki,
with a big thank you for your help. :)

**Comment by stsp** [comment]
Moved to a forum.

**Comment by stuaxo** [comment]
Would switching to something like Sphinx help for docs ?

Docs can be .rst files, then output to HTML (and PDF, though it's a bit more tricky).

There are also lots of examples of integrating with CI or use readthedocs.org to follow the repo.

As a proof of concept it would be fairly straightforward to convert the existing docs to .rst and setup a Makefile to generate HTML from it.

With some work docs can be styled - for instance https://docs.shoebot.net/  is generated from sphinx for instance and has slightly different style.

https://pycairo.readthedocs.io/en/latest/

**Comment by stsp** [comment]
We already have docbook docs that
are converted to html. I don't like that
conversion stuff: you need to either
keep html versions in sync or you can
remove html versions from git and always
rebuild them, but thats time-consuming.
Also writing such docs is challenging as
you need to always build them to check
if everything is ok.
Markdown is what I like so far, but certainly
there may be better options.
In any case, the problem is not to find
the doc format. The problem is to find the
doc writer. :)

**Comment by uralets** [comment]
As a follow up on the GENERAL MIDI issue.

What DOSEMU does in the context of using the SYSTEM MIDI facility is accessing something called ALSA Raw MIDI (no idea what's the diff between alsa_midi and alsa_virmidi in this context, again the issue with documentation... even though at this point I had to scour the sources).

While what's DOSBOX (and I presume the likes of SCUMMVM or aplaymidi) is using is accessing the ALSA *MIDI SEQUENCER*.
Now, maybe getting some actual SOUND with the scheme DOSEMU uses is simply a question of somehow hooking up this rawmidi thing to the actual MIDI port of the OS (Linux) that will produce the actual sound with aconnect (I admit I ended up not testing this out), but I can't see why it's not advantageous to DOSEMU to actually implement the same scheme as mentioned above (judging by the DOSBOX sources it's not complicated at all, no more so than this RAW MIDI thing you're opening now, DOSBOX even guesses the MIDI client:port - aka the 65:0 thing the system provides).
So it actually uses the system-wide sequencer by default with whatever soundfont is loaded into it.

As per now the MIDI sound coming out of DOSEMU is horrible (with whatever it is using internally to render MIDI), while what I'm getting of DOSBOX/SCUMMVM/aplaymidi with my system MIDI player and the custom MIDI font is noticeably better.

**Reply by stsp** [reply]
If you don't like the sound-font
that dosemu2 uses, then change
the /usr/share/soundfonts/default.sf2
symlink to whatever font you want.

If you want virmidi to work, then do
`sudo modprobe snd-virmidi`
after which, in `aconnect -l` you'll
find:
```
client 28: 'Virtual Raw MIDI 3-0' [type=kernel,card=3]
    0 'VirMIDI 3-0     '
client 29: 'Virtual Raw MIDI 3-1' [type=kernel,card=3]
    0 'VirMIDI 3-1     '
client 30: 'Virtual Raw MIDI 3-2' [type=kernel,card=3]
    0 'VirMIDI 3-2     '
client 31: 'Virtual Raw MIDI 3-3' [type=kernel,card=3]
    0 'VirMIDI 3-3  
```

which you can connect to any
synthesizer you like.

I think you are assuming much more
problems than actually do exist.

**Reply by stsp** [reply]
I added `$_fluid_sfont` now
so that you don't have to change
any symlinks.

**Reply by stsp** [reply]
OK, things were not all that
simple. Sorry, I wrote it 15 years
ago and completely forgot how
it works (and it didn't, which is
why alsa_virmidi was never
announced in dosemu.conf).

So to get alsa midi working, you
need something like this:
```
$_midi_driver = "alsa_midi,alsa_virmidi"
$_snd_plugin_params = "alsa_midi:dev_name=hw:3,0 alsa_virmidi:dev_name=hw:3,1"
```
And also you need the current
git code as I pushed the patches
to get that to work...
I documented `$_midi_driver`
setting properly in dosemu.conf
now. As for the `$_snd_plugin_params` -
it is only needed if your rawmidi
is not configured properly in
/etc/asound.conf. You need to
run `amidi -l` to see the virmidi
devices. If it lists nothing, then
`sudo modprobe snd-virmidi`.
Otherwise:
```
$ amidi -l
Dir Device    Name
IO  hw:3,0    Virtual Raw MIDI (16 subdevices)
IO  hw:3,1    Virtual Raw MIDI (16 subdevices)
IO  hw:3,2    Virtual Raw MIDI (16 subdevices)
IO  hw:3,3    Virtual Raw MIDI (16 subdevices)
```

As we see, the device is `hw:3,x`.
So either we hardcode that into
`$_snd_plugin_params` as shown
above, or we add this to /etc/asound.conf:
`defaults.rawmidi.card 3` and get the
rawmidi properly configured.

So yes, I admit the things may go
quite complex if you try to use some
non-default settings like the direct alsa
midi support is. I guess this is what
@runderwo was trying to tell in #117 :)

**Reply by stsp** [reply]
So after all the above, we need
to run `fluidsynth`. Then we see
`aconnect -l`:
```
$ aconnect -l
client 0: 'System' [type=kernel]
    0 'Timer           '
	Connecting To: 142:0
    1 'Announce        '
	Connecting To: 142:0
client 14: 'Midi Through' [type=kernel]
    0 'Midi Through Port-0'
client 28: 'Virtual Raw MIDI 3-0' [type=kernel,card=3]
    0 'VirMIDI 3-0     '
client 29: 'Virtual Raw MIDI 3-1' [type=kernel,card=3]
    0 'VirMIDI 3-1     '
client 30: 'Virtual Raw MIDI 3-2' [type=kernel,card=3]
    0 'VirMIDI 3-2     '
client 31: 'Virtual Raw MIDI 3-3' [type=kernel,card=3]
    0 'VirMIDI 3-3     '
client 128: 'FLUID Synth (44955)' [type=user,pid=44955]
    0 'Synth input port (44955:0)'
client 142: 'PipeWire-System' [type=user,pid=2121]
    0 'input           '
	Connected From: 0:1, 0:0
client 143: 'PipeWire-RT-Event' [type=user,pid=2121]
    0 'input           '
```

We note that fluidsynth allocated
port 128. So we do `aconnect 28:0 128:0`.
This is already enough to get the
alsa music in TerraNova game.
But we can also run munt and
connect the port "28:1" to it - then
we also get MT32 working.

**Reply by stsp** [reply]
But I guess its not documented
how do we get mt32 working...
which needs to be set up with
`emusound` tool.
So yes, I suppose the midi topic
needs a wiki page.

**Comment by uralets** [comment]
why not just implement the direct accessing of ALSA sequencer, like
other software does? And let the host system render midi through
whatever is hooked as the default sequencer (fluid/timi/real midi)...
Judging from DOSBOX sources it's not complicated at all (no more than
this RAW MIDI thing you currently have).
You admit you've forgotten the ins and outs of all this over the years,
why just not make it more straightforward for the end user..

As a side note I (vaguely) remember setting up the old dosemu1 and
since it didn't know anything about ALSA I had to go through some
arcane setup of all these virtual connecting but as an added difficulty
it had to read the output from a file where dosemu1 would write the
MIDI stream...

**Reply by stsp** [reply]
> why just not make it more straightforward for the end user..

You mean, you?
Other end users just use
the default config. People
who seek troubles, will find
them, but the wiki page may
still make sense, at least for
mt32 setup.

> You admit you've forgotten the ins and outs of all this over the years,

Because I can't remember all
possible advanced setup that
no one is going to use.

**Reply by stsp** [reply]
Of course the lack of an
option to alter soundfont,
was an omission.
Its now there.

**Reply by stsp** [reply]
https://github.com/dosemu2/dosemu2/wiki/MIDI-setup
MIDI documentation written.
HTH.

**Comment by rouyng** [comment]
Hi folks. I'm currently configuring dosemu2 to run a legacy business application and exploring the existing documentation in the process. I support uralets' concerns that the current documentation situation is rather scattered and incomplete. I'm not particularly concerned with ALSA MIDI or other specialized functionality (yet), but I do have some thoughts as dosemu2 novice looking to utilize the the project. I'd like to offer the following comments on the project's documentation:

1. As a new user, there's a bit of detective work involved in finding the dosemu2 docs. The main landing page for the project (first result on google, DDG, etc) is the github pages site at https://dosemu2.github.io/dosemu2/ . I suggest adding a "Documentation" subheading to this site that points readers towards the wiki and any other documentation locations users should know. Improving discoverability of existing documentation would be a big improvement, and likely invite further contributions to the wiki, etc. I'd be happy to open a PR for this myself, but the source repo for that github pages site is not immediately obvious.
2. Usually when I'm first getting familiar with a software package, the docs directory is a good first stop. However, currently the dosemu2 docs subdirectory is a bit of a grab bag of various files of unclear relevance in various formats. Many (all?) files under the docs subdirectory of the dosemu2 repository seem to be carried over from dosemu. Are these files in docs still relevant? How are they meant to be accessed (considering the various file formats)? Would it be advisable to convert them to markdown and/or integrate them into the wiki? 
3. Like the docs directly, the man page is also one of my first stops in understanding a new software package. The man page included with dosemu2 appears to be carried over from dosemu (unless something has gone wrong with my installation) and no longer seems to be very relevant to the current state of the project.  I'm not just coming in to ask for a new man page, as I know that takes a significant amount of work. However, IMO it would be an improvement to add sort of placeholder that states the original dosemu man page is deprecated and points to some of the actively maintained documentation locations, such as the usage info from `--help` or even the wiki. The current state is nearly worse than no man page because it is misleading.
4. I'd like to make some edits to the wiki to add some organization, some small copyedits, as well as include some existing documentation that currently lives in other places (such as in README.md, the usage information from `--help`, and possibly even from the docs folder if anything there is relevant/usable). Is there an established process for proposing/approving edits to the wiki? Or should I just go for it? I quite like this sort of documentation writing and editing as I do a lot in my day job, on my personal site, various specialty forums and Wikipedia. So I'm happy to contribute to the docs as I use the project, but I want to respect the maintainers' decisions about the proper process to do so.

**Reply by stsp** [reply]
> Hi folks. I'm currently configuring dosemu2 to run a legacy business application and exploring the existing documentation in the process.

Please, this doesn't tell us
a tiny bit about what problem
are you trying to solve.
Why do you "configure dosemu2
for legacy application"? Does
this application crash? Does
dosemu2 crash? What problem?

> I support uralets' concerns that the current documentation situation is rather scattered and incomplete.

Concerns are not counted.
Problems are counted.
uralets provided a gfx problem
(in another ticket) and that needs
to be fixed, but documentation
won't help with that.

> As a new user, there's a bit of detective work involved in finding the dosemu2 docs.

Because you are supposed to run
dosemu2 first, and only look for
solutions once you encounter a
problem. Not before.
README.md contains everything
you need to start working.

> I'd be happy to open a PR for this myself, but the source repo for that github pages site is not immediately obvious.

Source repo is the same but
the branch is called `gh-pages`.

> Are these files in docs still relevant?

Some are, for example in `tweaks.html`
there is a collection of recipes to
run the particular problematic programs.

> The man page included with dosemu2 appears to be carried over from dosemu

Because dosemu2 maintains some
compatibility with dosemu1 wrt
command-line. And in fact, I am
trying to keep the man page up
to date. If you find any problem
with man page - don't hesitate to
fill a bug or correct it and send PR.
`--help` should probably just suggest
to run `man dosemu` and/or say
something else very briefly.
So if you are really at it, please
consider changing `--help` to not
duplicate the man page but rather
to refer to it.

> Or should I just go for it?

Yeah, feel free to edit wiki.
We don't have too many doc writers
to set a "high standards" for a
prospective doc writers. :) I don't
even know if I'll be notified once you
do the edits. If not - that's bad.

**Reply by rouyng** [reply]
Hi stsp, thank you for the quick reply. 

> Please, this doesn't tell us
> a tiny bit about what problem
> are you trying to solve.
> Why do you "configure dosemu2
> for legacy application"? Does
> this application crash? Does
> dosemu2 crash? What problem?

That was just to introduce myself and why I am interested in the project. I'm not having a problem with dosemu2. I'm just posting here because I would like to contribute to the documentation.

> Because you are supposed to run
> dosemu2 first, and only look for
> solutions once you encounter a
> problem. Not before.
> README.md contains everything
> you need to start working.

I don't share your opinion that documentation is not needed until a user encounters problems. I think this depends on an individual person's learning approach. Your style will work for some people, while other people will prefer to "RTFM" before diving in. However, having more/better organized documentation will benefit both folks that like your approach AND folks like me that want to read docs first. So it's not worth arguing about :)

Thank you for the instructions about contributing and the explanation of your approach to the project's documentation. I'm going to go ahead and make some contributions to the wiki in the next few days. Basically, I would like to put together a "table of contents" for the wiki's front page, which will organize it for future expansion. I would also like to add additional pages for some documentation that already exists in README.md, tweaks.html and the man page. That way there is one "entry point" for the project's documentation, at risk of duplicating some content.  I'll also put together a pull request for the `gh-pages` branch to add a documentation subheading with a link. I don't think this will take very long, but it will hopefully help people getting started with dosemu2. If you would like, I can ping you in this discussion thread once I've made my edits, so you you will get a notification.

**Reply by stsp** [reply]
> I'm just posting here because I would like to contribute to the documentation.

This is more than welcome.
But if you want to write some documentation
(rather than to just do a minor corrections
to an existing one), then you should first
run dosemu2 and get used to it.

> other people will prefer to "RTFM" before diving in.

But what kind of RTFM should
that be? You already have README.md
for a quick start. What exactly RTFM do
you need before starting? dosemu2
history? World War 2 history? Please
name the topics you need to read
before starting, and lets consider
adding them to README.md.

> I can ping you in this discussion thread once I've made my edits, so you you will get a notification.

Please do and we'll see if auto
notifications do work or not.

**Reply by rouyng** [reply]
I made some edits to the wiki, let me know your thoughts. Biggest change was the [home page](https://github.com/dosemu2/dosemu2/wiki), where I added some categories and links to existing documentation and resources. That way the wiki home page can quickly direct people to the information they need. Also, the categories will help organize any future documentation people would like to write.

I also converted doc/tweak.html to markdown and added it to [a wiki page](https://github.com/dosemu2/dosemu2/wiki/Tweaks). I also added a link to fddp on [Building from source](https://github.com/dosemu2/dosemu2/wiki/Building-From-Source) and a link to openSUSE binary repos on [Download sites for dosemu2](https://github.com/dosemu2/dosemu2/wiki/Download-sites-for-dosemu2). Just small changes but it makes it easier to find the documentation and resources that already exist.

> This is more than welcome.
> But if you want to write some documentation
> (rather than to just do a minor corrections
> to an existing one), then you should first
> run dosemu2 and get used to it.

Yes, I agree and I plan to do that. However, IMO there are easy improvements possible just by editing existing documentation to make things more discoverable. That's what I tried to do with the wiki edits. Next I will make a PR for adding a documentation heading and wiki link on the GitHub pages site.

**Reply by stsp** [reply]
Wow, you are cool!
Many thanks. :)
I did a small edits myself now
too. Also I am very annoyed by
the fact that the notifications
from wiki are not coming, but
that seems to be unimplemented
on github so nothing to do.

> I also converted doc/tweak.html to markdown and added it to a wiki page

Great!
Maybe now I should clean up
doc/README.html which contains
both old and new stuff, and after
that convert it to md, and remove
the in-tree docs completely!

Does anyone object against making
`--help` to just state the license and
refer to a man pages?
I think if we want to get proper docs,
we need to first make them terse. :)

**Reply by rouyng** [reply]
You are very welcome. As I said, I enjoy this kind of work, when I have the time.

My opinions:

> Great!
> Maybe now I should clean up
> doc/README.html which contains
> both old and new stuff, and after
> that convert it to md, and remove
> the in-tree docs completely!

I think cleaning up obsolete/deprecated information from docs like doc/README.html and the man pages would be a huge improvement. Currently, it's hard to "trust" those documents when it's unclear what information is no longer relevant to the current state of the project. doc/README.html in particular seems like a great document, if I knew what info still applied to dosemu2. The clean-up should be done by someone like you, who has an intimate knowledge of the current development state of dosemu2. However, I can volunteer to do the markdown conversion and any needed copy editing, if that helps you. 

The follow-up question is how should docs/README.html be incorporated into the wiki structure? My opinion is that each major section of that document (numbered 1 through 15) could be a separate wiki page. Each section can be read/understood mostly independently. They can categorized under the "Running" or "Configuation" sections of the wiki, depending on the topic. The benefit of having each section as a separate page is that it can be easily searched and categorized using existing wiki functionality (easier for users to find). Also IMO, smaller separate pages are easier for wiki contributors to edit and continuously improve. Again, I am happy to assist with making/categorizing these pages once the original document is brought up to date.

> Does anyone object against making
> --help to just state the license and
> refer to a man pages?
> I think if we want to get proper docs,
> we need to first make them terse. :)

I agree that brevity is a virtue, especially when trying to remove outdated info and keep docs current. However, having a usage statement invoked by "--help" is a very strong Linux convention, used by almost every command line tool. Having a terse description of the positional arguments and option flags is very useful, especially for a program like dosemu2 which has 37 (by my count) option flags. For example, I may forget whether the option I need is invoked by "-e" or "-E". Running "dosemu2 --help" is a much faster way to answer that question than scrolling through man pages. The output of "--help" fits in one screen and can be scanned quickly, while the man page usually requires several screens of scrolling and scanning to find the answer. So, it is an additional maintenance burden to keep "--help" up to date, but I believe it is worth it, especially because most Linux users expect to access usage information that way. To be clear, I don't believe "--help" needs to be expanded, but the current state is good in my opinion.

**Comment by stsp** [comment]
> However, I can volunteer to do the markdown conversion and any needed copy editing, if that helps you.

OK I reviewed the README.html
and there is hardly a single section
that still applies. But not too many
sections are entirely useless and
needs to be removed - no, most
really have to be updated, as the
topics they cover, may still be of
an interest.
So I evaluated the amount of needed
work, and its quite big. And also got
an impression how big the project is,
overall... And there are also other
docs to review, namely:
- EMUfailure.html
- NOVELL-HOWTO.txt
- DANG.sgml
- dosemu-HOWTO.sgml

> My opinion is that each major section of that document (numbered 1 through 15) could be a separate wiki page.

Indeed.
Anyway, you can start from converting
the README chapter 9 to md, that is,
a chapter about win31. It seems more
or less relevant. Lets see how it goes.

**Reply by rouyng** [reply]
Ok, great! [Here's chapter 9 as a wiki page.](https://github.com/dosemu2/dosemu2/wiki/Running-Windows-under-DOSEMU) Ping me here if you review any more.

**Reply by stsp** [reply]
Thanks!
I've done the needed changes.
@jschwartzenberg would you like
to update the trident section? IIRC
you found some better drivers.

> Ping me here if you review any more.

Well I think the process can work
better if you first convert and then
we review. :)
For example you can convert
section 1.1 and I'll update it online
later.

**Reply by rouyng** [reply]
Oh yeah, that makes sense.
I did ["DOSEMU modes of operation"](https://github.com/dosemu2/dosemu2/wiki/DOSEMU-modes-of-operation) (section 1.1) and ["Runtime Configuration Options"](https://github.com/dosemu2/dosemu2/wiki/Runtime-Configuration-Options) (section 2).

One issue I noticed in README.html is that there are a few references to a missing README-tech.txt file.

**Reply by stsp** [reply]
> I did

Thanks!
I unlinked them from the main page
for now as they are not up to date.
But its a good stating point.
I hope other people can now look
into updating those too, as that now
requires much fewer work and can
be done sequentially.

> One issue I noticed in README.html is that there are a few references to a missing README-tech.txt file.

It seems, most of the dosemu1
documentation was removed by
e60d6b9b.
README-tech.sgml configuration.sgml sound-usage.sgml
were removed.
You can restore it with this command:
`git checkout e60d6b9b^ doc/README-tech.txt`
I wonder if someone finds such way
of documenting the internal architecture,
useful. I admit that dosemu2 architecture
is 100 times more complex and not
documented at all, but I think writing
a half-MB doc like README-tech is
just not going to help. Yet it would be
interesting to find some useful way of
documenting things, that would allow
the new/former devs to not get alienated
as quickly as now.
